internal/storage/storage_handle.go

```diff
@@ -61,18 +61,26 @@ func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.Stora
 
 	// Add Custom endpoint option.
 	if clientConfig.CustomEndpoint != nil {
-		clientOpts = append(clientOpts, option.WithEndpoint(storageutil.StripScheme(clientConfig.CustomEndpoint.String())))
-		// Explicitly disable auth in case of custom-endpoint, aligned with the http-client.
-		// TODO: to revisit here when supporting TPC for grpc client.
-		clientOpts = append(clientOpts, option.WithoutAuthentication())
-		clientOpts = append(clientOpts, option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())))
+		if clientConfig.AnonymousAccess {
+			clientOpts = append(clientOpts, option.WithEndpoint(storageutil.StripScheme(clientConfig.CustomEndpoint.String())))
+			// Explicitly disable auth in case of custom-endpoint, aligned with the http-client.
+			// TODO: to revisit here when supporting TPC for grpc client.
+			clientOpts = append(clientOpts, option.WithoutAuthentication())
+			clientOpts = append(clientOpts, option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())))
+		} else {
+			logger.Fatal("GRPC path is not supported with authentication on custom endpoint.")
```

Comment:
nit: `GRPC client doesn't support auth for custom-endpoint. Please set anonymous-access: true via config-file.`

Also, do we see this error while mounting? What happens while mounting in background mode.

Reply:
> do we see this error while mounting?

Yes

> What happens while mounting in background mode.

It requires log file.

Reply:
return error instead of fatal. Now it will be visible in background process also.